search-service/src/main/kotlin/com/egm/stellio/search/authorization/EntityAccessRights.kt

```diff
@@ -68,3 +78,67 @@ data class EntityAccessRights(
         return resultEntity
     }
 }
+
+/**
+ * Build the expanded subject info.
+ *
+ * For instance:
+ *
+ * "[
+ *   {
+ *     "@type": [
+ *       "https://uri.etsi.org/ngsi-ld/Property"
+ *     ],
+ *     "https://uri.etsi.org/ngsi-ld/hasValue": [
+ *       {
+ *         "https://ontology.eglobalmark.com/authorization#kind": [
+ *              {
+ *                  "@value": "kind"
+ *               }
+ *         ],
+ *         "https://ontology.eglobalmark.com/authorization#username": [
+ *              {
+ *                  "@value": "username"
+ *              }
+ *         ]
+ *       }
+ *     ]
+ *   }
+ * ]
+ */
+fun buildExpandedSubjectInfo(value: Map<String, String>): ExpandedAttributePayload =
+    listOf(
+        mapOf(
+            JSONLD_TYPE to listOf(NGSILD_PROPERTY_TYPE.uri),
+            NGSILD_PROPERTY_VALUE to listOf(
```

Comment:
why don't you just iterate over the map of subject info instead of hardcoding keys (and missing some entries btw)?

Reply:
Yeah but I don't see how expand keys. 
Currently the keys are kind and username (name or subjectId) but I don't see how to put them in expanded format.

Reply:
what if you simply use `expandJsonLdFragment` to expand the whole `subjectInfo`?